src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -295,25 +278,15 @@ export function filterTelemetryCacheEvents(input: any): TelemetryEvent[] {
             return true
         })
         .filter((item: Object) => {
-            // Only accept objects that have data because that's what's required by TelemetryEvent
-            if (!item.hasOwnProperty('data')) {
-                getLogger().warn(`Item in telemetry cache: ${item}\n does not have 'data' or 'createTime'! skipping!`)
-
-                return false
-            }
-
-            return true
-        })
-        .filter((item: TelemetryEvent) => {
             // skip it if data is not an array or empty
-            if (!Array.isArray(item.data) || item.data.length === 0) {
-                getLogger().warn(`Item in telemetry cache: ${item}\n has invalid data field: ${item.data}! skipping!`)
+            if (!Array.isArray(item) || item.length === 0) {
```

Comment:
do we need a test to make sure we gracefully handle cached entries created before this change?

Reply:
we gracefully handle them by throwing them out, we have 4 generations of this cache so far and have never retained the old versions. The test will be updated to make sure it handles it properly.